packages/snap/src/rpc/signTransactions.ts

```diff
@@ -22,8 +23,30 @@ export async function signTransactions(
 
   const signer = new InMemorySigner(keystore);
 
-  for (const transaction of transactionsArray) {
+  const provider = new providers.JsonRpcProvider({
+    url: `https://rpc.${params.network}.near.org`,
+  });
+
+  // fetch data from chain - the latest block hash and latest nonce of account
+  const accessKey = await provider.query(
+    `access_key/${Buffer.from(keyPair.getPublicKey().data).toString(
+      "hex"
+    )}/${keyPair.getPublicKey().toString()}`,
+    ""
+  );
+  let nonce = ++(accessKey as unknown as { nonce: number }).nonce;
```

Comment:
caller does not supply nonce?

Reply:
by the sow caller only send `receiverId`  and list of `actions` if we are going to change that caller send this data then we need to add filed `nonce` and `recentBlockHash`

Reply:
what does near wallet selector send?

Reply:
```ts
export interface SignAndSendTransactionParams {
  signerId?: string;
  receiverId?: string;
  actions: Array<Action>;
}
```
but we can use `provider.query` to get the latest `nonce` and `recentBlockHash` (from signAndSendTransaction) from wallet-selector and make snap more as Keystore and singing tool

Reply:
Let's allow optional nonce and recentBlockHash params to be sent. And if they aren't, than use provider.

Also can you create sign and signAndSend methods so we can decide later on in near-wallet-selector what is better.

FYI check note and document how are we handling it: https://github.com/near/wallet-selector/blob/main/packages/core/docs/guides/custom-wallets.md#signandsendtransactions